data-tiles/GNUmakefile

```diff
@@ -0,0 +1,355 @@
+#
+# This makefile requires GNU make
+#
+# Important targets:
+#
+#	all -- do everything
+#
+#	dirs -- create data directory structure
+#	binaries -- compile Go programs (you may need -B)
+#
+#	download -- download geojson and metrics files
+#	extract -- extract CSVs from metrics zip files
+#	standard -- normalise downloaded geojson (add bbox, set geotype, etc)
+#
+#	geos -- generate geojson files for every geography in standard geojson
+#	metrics -- split downloaded CSVs into single-metric files
+#	tiles - generate data tile CSVs
+#	breaks -- generate break
+#
+#	clean -- remove generated files
+#	realclean -- remove generated and downloaded files
+
+.DEFAULT_GOAL: all
+
+#
+# Files to download
+#
+
+# Location of nomis zips
+NOMIS_URL=https://www.nomisweb.co.uk/output/census/2011
+
+# List of nomis zips to download, taken from Viv's v4 google spreadsheet.
+NOMIS_ZIPS=\
+	ks103ew_2011_oa.zip \
+	ks202ew_2011_oa.zip \
+	ks206ew_2011_oa.zip \
+	ks207wa_2011_oa.zip \
+	ks608ew_2011_oa.zip \
+	qs101ew_2011_oa.zip \
+	qs103ew_2011_oa.zip \
+	qs104ew_2011_oa.zip \
+	qs113ew_2011_oa.zip \
+	qs119ew_2011_oa.zip \
+	qs201ew_2011_oa.zip \
+	qs202ew_2011_oa.zip \
+	qs203ew_2011_oa.zip \
+	qs208ew_2011_oa.zip \
+	qs301ew_2011_oa.zip \
+	qs302ew_2011_oa.zip \
+	qs303ew_2011_oa.zip \
+	qs402ew_2011_oa.zip \
+	qs403ew_2011_oa.zip \
+	qs406ew_2011_oa.zip \
+	qs411ew_2011_oa.zip \
+	qs415ew_2011_oa.zip \
+	qs416ew_2011_oa.zip \
+	qs501ew_2011_oa.zip \
+	qs601ew_2011_oa.zip \
+	qs604ew_2011_oa.zip \
+	qs605ew_2011_oa.zip \
+	qs701ew_2011_oa.zip \
+	qs702ew_oa.zip \
+	qs803ew_2011_oa.zip \
+
+# S3 prefix for downloading raw geojson files (in dp-sandbox environment)
+S3_URL=s3://ons-dp-sandbox-atlas-input/geojson
+
+# Raw LAD geojson
+RAW_LAD=Local_Authority_Districts_(December_2017)_Boundaries_in_the_UK_(WGS84).geojson
+
+# Raw LSOA geojson
+RAW_LSOA=Lower_Layer_Super_Output_Areas_(December_2011)_Boundaries_Super_Generalised_Clipped_(BSC)_EW_V3.geojson
+
+# Raw MSOA geojson
+RAW_MSOA=Middle_Layer_Super_Output_Areas_(December_2011)_Boundaries_Super_Generalised_Clipped_(BSC)_EW_V3.geojson
+
+# Raw OA geojson
+RAW_OA=Output_Areas__December_2011__Boundaries_EW_BGC.geojson
+
+# Name of CSV holding MSOA names
+MSOA_NAMES=MSOA-Names-1.16.csv
+
+# URL to download MSOA names
+MSOA_URL=https://houseofcommonslibrary.github.io/msoanames/$(MSOA_NAMES)
+
+#
+# Directory locations
+#
+D=data
+DD=$D/downloads
+DDG=$(DD)/geo
+DDM=$(DD)/metrics
+DP=$D/processed
+DPG=$(DP)/geo
+DPM=$(DP)/metrics
+DO=$D/output
+DOG=$(DO)/geo
+DOB=$(DO)/breaks
+DOT=$(DO)/tiles
+
+
+# All the raw geojson files to download
+RAW_GEOJSON=$(RAW_LAD) $(RAW_LSOA) $(RAW_MSOA) $(RAW_OA)
+
+#
+# do everything
+#
+.PHONY: all
+all: dirs $(DOG)/.done $(DOT)/.done $(DOB)/.done
```

Comment:
probably missing something here, but just FMI how does this target work? it looks like it just makes some directories to me?

Reply:
Further on in the makefile, the .done files are targets with their own dependencies.

Reply:
ahhh they're dynamically-named targets? sweet!